sdk/python/kfp/compiler/_component_builder.py

```diff
@@ -93,90 +93,56 @@ def generate_pip_requirements(self, target_file):
+      filename (str): requirement file name, default as requirements.txt
+  """
+  dependency_helper = DependencyHelper()
+  for version in dependency:
+    dependency_helper.add_python_package(version)
+  dependency_helper.generate_pip_requirements(filename)
 
-  def _generate_dockerfile_with_py(self, target_file, base_image, python_filepath, has_requirement_file, python_version):
-    """ _generate_docker_file generates a simple dockerfile with the python path
-    args:
-      target_file (str): target file name for the dockerfile.
+def _generate_dockerfile(filename, base_image, entrypoint_filename, python_version, requirement_filename=None):
+  """
+    generates dockerfiles
+    Args:
+      filename (str): target file name for the dockerfile.
       base_image (str): the base image name.
-      python_filepath (str): the path of the python file that is copied to the docker image.
-      has_requirement_file (bool): whether it has a requirement file or not.
+      entrypoint_filename (str): the path of the entrypoint source file that is copied to the docker image.
       python_version (str): choose python2 or python3
-    """
-    if python_version not in ['python2', 'python3']:
-      raise ValueError('python_version has to be either python2 or python3')
-    with open(target_file, 'w') as f:
-      f.write('FROM ' + base_image + '\n')
-      if python_version is 'python3':
-        f.write('RUN apt-get update -y && apt-get install --no-install-recommends -y -q python3 python3-pip python3-setuptools\n')
-      else:
-        f.write('RUN apt-get update -y && apt-get install --no-install-recommends -y -q python python-pip python-setuptools\n')
-      if has_requirement_file:
-        f.write('ADD ' + self._ARC_REQUIREMENT_FILE + ' /ml/\n')
-        if python_version is 'python3':
-          f.write('RUN pip3 install -r /ml/' + self._ARC_REQUIREMENT_FILE + '\n')
-        else:
-          f.write('RUN pip install -r /ml/' + self._ARC_REQUIREMENT_FILE + '\n')
-      f.write('ADD ' + python_filepath + " /ml/" + '\n')
+      requirement_filename (str): requirement file name
+  """
+  if python_version not in ['python2', 'python3']:
+    raise ValueError('python_version has to be either python2 or python3')
+  with open(filename, 'w') as f:
+    f.write('FROM ' + base_image + '\n')
+    if python_version is 'python3':
+      f.write('RUN apt-get update -y && apt-get install --no-install-recommends -y -q python3 python3-pip python3-setuptools\n')
+    else:
+      f.write('RUN apt-get update -y && apt-get install --no-install-recommends -y -q python python-pip python-setuptools\n')
+    if requirement_filename is not None:
+      f.write('ADD ' + requirement_filename + ' /ml/requirements.txt\n')
       if python_version is 'python3':
-        f.write('ENTRYPOINT ["python3", "/ml/' + python_filepath + '"]')
+        f.write('RUN pip3 install -r /ml/' + requirement_filename + '\n')
```

Comment:
it should be always /ml/requirements.txt, right?

Reply:
in the image, it will always be /ml/requirements.txt
However, the requirement_filename might not be. This internal function might be used by some other module that does not have a hardcoded name.

Reply:
Correct. I misunderstood.
fixed now.